converter.go

```diff
@@ -191,25 +193,43 @@ func MakeStringToDatetimeConverter() StringToDatetimeConverter {
 }
 
 const (
-	ttTimeLayout   = "2006-01-02T15:04:05.999999-0700"
-	dateTimeLayout = "2006-01-02 15:04:05"
-	dateOnlyLayout = "2006-01-02"
+	dateLayout           = "2006-01-02"
+	dateTimeLayout       = "2006-01-02T15:04:05.999999999"
+	dateTimeOffsetLayout = "2006-01-02T15:04:05.999999999-0700"
 )
 
 // Convert is the implementation of Converter[string, any] for StringToDatetimeConverter.
 func (StringToDatetimeConverter) Convert(src string) (any, error) {
+	// Try to parse without TZ name.
 	for _, layout := range [...]string{
-		time.RFC3339,
+		dateLayout,
 		dateTimeLayout,
-		dateOnlyLayout,
-		ttTimeLayout,
+		dateTimeOffsetLayout,
 	} {
 		result, err := time.Parse(layout, src)
 		if err != nil {
 			continue
 		}
-		return datetime.NewDatetime(result.UTC())
+		_, offset := result.Zone()
+		result = result.In(time.FixedZone(datetime.NoTimezone, offset))
+		return datetime.NewDatetime(result)
 	}
+
+	// Try to parse with TZ name.
+	if date, tzName, ok := strings.Cut(src, " "); ok {
```

Comment:
Can we move "try to parse without timezone" after this block, or under `else`. No need to try to parse datetime string using different layouts if tz is specified.